src/equations/hyperbolic_serre_green_naghdi_1d.jl

```diff
@@ -0,0 +1,364 @@
+    + \biggl( \frac{\lambda}{3} \eta (1 - \eta / h) \biggr)_x
+    + \biggl( g h + \frac{\lambda}{2} (1 - \eta / h) \biggr) b_x &= 0,\\
+  h w_t + h v w_x &= \lambda (1 - \eta / h),\\
+  \eta_t + \eta_x u + \frac{3}{2} b_x v &= w.
+\end{aligned}
+```
+
+References for the hyperbolized Serre-Green-Naghdi system can be found in
+- Favrie and Gavrilyuk.
+  A rapid numerical method for solving Serre-Green-Naghdi equations
+  describing long free surface gravity waves
+  [DOI: 10.1088/1361-6544/aa712d](https://doi.org/10.1088/1361-6544/aa712d)
+- Busto, Dumbser, Escalante, Favrie, and Gavrilyuk.
+  On High Order ADER Discontinuous Galerkin Schemes for First Order Hyperbolic
+  Reformulations of Nonlinear Dispersive Systems
+  [DOI: 10.1007/s10915-021-01429-8](https://doi.org/10.1007/s10915-021-01429-8)
+
+The semidiscretization implemented here conserves
+- the total water mass (integral of h) as a linear invariant
+- the total modified energy
+
+for periodic boundary conditions (see Ranocha and Ricchiuto (2024)).
+Additionally, it is well-balanced for the lake-at-rest stationary solution, see
+- Hendrik Ranocha and Mario Ricchiuto (2024)
+  Structure-preserving approximations of the Serre-Green-Naghdi
+  equations in standard and hyperbolic form
+  [arXiv: 2408.02665](https://arxiv.org/abs/2408.02665)
+"""
+struct HyperbolicSerreGreenNaghdiEquations1D{Bathymetry <: Union{BathymetryFlat, BathymetryMildSlope}, RealT <: Real} <:
+       AbstractSerreGreenNaghdiEquations{1, 5}
+    bathymetry_type::Bathymetry # type of bathymetry
+    gravity::RealT # gravitational constant
+    eta0::RealT # constant still-water surface
+    λ::RealT # hyperbolic relaxation parameter (→ ∞ for Serre-Green-Naghdi)
+end
+
+function HyperbolicSerreGreenNaghdiEquations1D(bathymetry_type = bathymetry_mild_slope;
+                                               gravity_constant,
+                                               eta0 = 0.0,
+                                               λ)
+    HyperbolicSerreGreenNaghdiEquations1D(bathymetry_type, gravity_constant, eta0, λ)
+end
+
+# TODO: How shall we handle eta=h+b vs the new variable η?
+function varnames(::typeof(prim2prim), ::HyperbolicSerreGreenNaghdiEquations1D)
+  return ("η", "v", "D", "w", "η")
+end
+function varnames(::typeof(prim2cons), ::HyperbolicSerreGreenNaghdiEquations1D)
+  return ("h", "hv", "b", "hw", "hη")
+end
+
+# TODO: There is another name clash. For the SerreGreenNaghdiEquations1D,
+#       the corresponding function is called initial_condition_convergence_test
+#       However, we cannot use that name since it's not an analytical solution.
+#       How shall we handle this?
+"""
+    initial_condition_soliton(x, t, equations::HyperbolicSerreGreenNaghdiEquations1D, mesh)
+
+A soliton solution of the [`SerreGreenNaghdiEquations1D`](@ref)
+used for convergence tests in a periodic domain.
+
+See also [`initial_condition_convergence_test`](@ref).
+"""
```

Comment:
Another name clash

Reply:
Good question. I would be fine keeping it like it is now (maybe with a bit more explanations in the docstring that this function is the same as `initial_condition_convergence_test` for `SerreGreenNaghdiEquations1D`, but is not an analytical solution of `HyperbolicSerreGreenNaghdiEquations1D`). I think it's not too bad that these two initial conditions are not called the same.

Reply:
Well, they represent the same physics, so it's a bit annoying but I don't know a better solution right now 🤷

Reply:
Yeah, it' not perfect, but I can live with in (at least for now). The name `initial_condition_convergence_test` doesn't _describe_ the physics, but rather the fact of it being an analytical solution of some equations, so it's not _that_ confusing, I think.